src/hostapi/alsa/pa_linux_alsa.c

```diff
@@ -1063,6 +1063,7 @@ static PaError PaAlsa_StrDup( PaAlsaHostApiRepresentation *alsaApi,
  */
 static int IgnorePlugin( const char *pluginId )
 {
+#ifndef PA_ALSA_IGNORE_ALL_PLUGINS
```

Comment:
Is there a reason why you did the #ifndef here rather than at the call site? :

https://github.com/PortAudio/portaudio/blob/e1f0c230aac0d9aa0d45b4f8600061de6f14c62f/src/hostapi/alsa/pa_linux_alsa.c#L1388

Maybe the debug logging should not be suppressed.

Reply:
I did not want to pollute the code using an existing `IgnorePlugin` function, in fact modifying `IgnorePlugin` is perfectly fine for the logic of `ifdef` as it already ignoring some plugins and with `#ifndef PA_ALSA_IGNORE_ALL_PLUGINS` will ignore all plugins, that is the goal of the proposed change.

Reply:
If we used a getenv() then we could just skip the entire block that starts at 1358.

Reply:
If we used a getenv() then we could just skip the entire block that starts at 1363.
Maybe something like:

    if (getenv("PA_ALSA_IGNORE_ALL_PLUGINS")
            && atoi( getenv( "PA_ALSA_IGNORE_ALL_PLUGINS" ) ) )
    {
                PA_DEBUG(( "%s: Ignoring all ALSA plugin devices\n", __FUNCTION__ ));
    }
    else if( ( res = alsa_snd_config_search( *alsa_snd_config, "pcm", &topNode ) ) >= 0 )

Reply:
@dmitrykos - What do you think about using getenv()?
Using getenv() makes it machine specific so users can configure it. Plugins are also machine specific so it seems more appropriate to use getenv().

Reply:
@philburk , thank you for your comment. I agree, will update implementation accordingly.